test/hls/hls_parser_unit.js

```diff
@@ -2745,4 +2748,151 @@ describe('HlsParser', () => {
+        '#EXT-X-MAP:URI="init.mp4",BYTERANGE="616@0"\n',
+        '#EXTINF:5,\n',
+        '#EXT-X-BYTERANGE:121090@616\n',
+        'main.mp4',
+      ].join('');
+
+      const manifest = shaka.test.ManifestGenerator.generate((manifest) => {
+        manifest.anyTimeline();
+        manifest.addPartialVariant((variant) => {
+          variant.addPartialStream(ContentType.AUDIO, (stream) => {
+            stream.mime('audio/mp4', 'mp4a');
+          });
+        });
+      });
+
+      await testHlsParser(master, media, manifest);
+
+      const eventValue = {
+        type: 'sessiondata',
+        id: 'fooId',
+        language: 'en',
+        value: 'fooValue',
+      };
+      expect(onEventSpy).toHaveBeenCalledWith(
+          jasmine.objectContaining(eventValue));
+    });
+
+    it('parses uri data', async () => {
+      const master = [
+        '#EXTM3U\n',
+        '#EXT-X-SESSION-DATA:DATA-ID="fooId",URI="foo.json"\n',
+        '#EXT-X-STREAM-INF:BANDWIDTH=200,CODECS="mp4a"\n',
+        'audio',
+      ].join('');
+
+      const media = [
+        '#EXTM3U\n',
+        '#EXT-X-PLAYLIST-TYPE:VOD\n',
+        '#EXT-X-MAP:URI="init.mp4",BYTERANGE="616@0"\n',
+        '#EXTINF:5,\n',
+        '#EXT-X-BYTERANGE:121090@616\n',
+        'main.mp4',
+      ].join('');
+
+      const manifest = shaka.test.ManifestGenerator.generate((manifest) => {
+        manifest.anyTimeline();
+        manifest.addPartialVariant((variant) => {
+          variant.addPartialStream(ContentType.AUDIO, (stream) => {
+            stream.mime('audio/mp4', 'mp4a');
+          });
+        });
+      });
+
+      await testHlsParser(master, media, manifest);
+
+      const eventValue = {
+        type: 'sessiondata',
+        id: 'fooId',
+        uri: 'test:/foo.json',
+      };
+      expect(onEventSpy).toHaveBeenCalledWith(
+          jasmine.objectContaining(eventValue));
+    });
+
+    it('parses mutiple data', async () => {
+      const master = [
+        '#EXTM3U\n',
+        '#EXT-X-SESSION-DATA:DATA-ID="fooId",LANGUAGE="en",VALUE="fooValue"\n',
+        '#EXT-X-SESSION-DATA:DATA-ID="fooId",LANGUAGE="es",VALUE="fooValue"\n',
+        '#EXT-X-SESSION-DATA:DATA-ID="fooId",URI="foo.json"\n',
+        '#EXT-X-STREAM-INF:BANDWIDTH=200,CODECS="mp4a"\n',
+        'audio',
+      ].join('');
+
+      const media = [
+        '#EXTM3U\n',
+        '#EXT-X-PLAYLIST-TYPE:VOD\n',
+        '#EXT-X-MAP:URI="init.mp4",BYTERANGE="616@0"\n',
+        '#EXTINF:5,\n',
+        '#EXT-X-BYTERANGE:121090@616\n',
+        'main.mp4',
+      ].join('');
+
+      const manifest = shaka.test.ManifestGenerator.generate((manifest) => {
+        manifest.anyTimeline();
+        manifest.addPartialVariant((variant) => {
+          variant.addPartialStream(ContentType.AUDIO, (stream) => {
+            stream.mime('audio/mp4', 'mp4a');
+          });
+        });
+      });
+
+      await testHlsParser(master, media, manifest);
+
+      expect(onEventSpy).toHaveBeenCalledTimes(3);
```

Comment:
You can also use toHaveBeenCalledWith() three times with the three expected payloads.  That will ensure that we don't somehow reuse the first tag three times.  (Similar bugs have occurred many times in async parts of the DASH parser.)

Reply:
I changed it!